File: transpiler/javatests/com/google/j2cl/readable/java/simpleliterals/SimpleLiterals.java
/*
 * Copyright 2017 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package simpleliterals;

public class SimpleLiterals {
  @SuppressWarnings("unused")
  public void main() {
    boolean bool = false;
    char ch = 'a';
    byte b = 127;
    byte nb = -128;
    short s = 32767;
    short ns = -32768;
    int i = 101;
    int ni = -101;
    long l = 101L;
    long nl = -101L;
    float f = 101.0f;
    float nf = -101.0f;
    float zf = 0.0f;
    float nzf = -0.0f;
    double d = 101.0;
    double nd = -101.0;
    double zd = 0.0;
    double nzd = -0.0;
    Object o = null;
    String str = "foo";
    Class<?> c = SimpleLiterals.class;
    float zeroF = -0.0f;
    float minusZeroF = -0.0f;
    double zeroD = 0.0;
    double minusZeroD = -0.0;
    double minusMinusZeroD = - -0.0;
  }
}

File: transpiler/javatests/com/google/j2cl/readable/kotlin/simpleliterals/SimpleLiterals.kt
/*
 * Copyright 2022 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package simpleliterals

class SimpleLiterals {
  fun main() {
    val bool = false
    val ch = 'a'
    val b: Byte = 127
    val nb: Byte = -128
    val s: Short = 32767
    val ns: Short = -32768
    val i = 101
    val ni = -101
    val l = 101L
    val nl = -101L
    val f = 101.0f
    val nf = -101.0f
    val zf = 0f
    val nzf = -0f
    val d = 101.0
    val nd = -101.0
    val zd = 0.0
    val nzd = -0.0
    val o: Any? = null
    val str = "foo"
    val c = SimpleLiterals::class
    val zeroF = 0.0f
    val minusZeroF = -0.0f
    val zeroD = 0.0
    val minusZeroD = -0.0
    val minusMinusZeroD = - /* see b/235149840 */ -0.0
  }
}
